test/cql-pytest/test_materialized_view.py

```diff
@@ -992,3 +992,86 @@ def test_base_clustering_prefix_deletion(cql, test_keyspace):
+        # The first view has the same partition key as the base, so the entire partition will be deleted on the view as well.
+        # The other views have different partition key than the base, so they will have individual rows removed.
+        with new_materialized_view(cql, table, '*', 'p,c', 'p is not null and c is not null', extra='with synchronous_updates = true') as mv1, \
+             new_materialized_view(cql, table, '*', 'c,p', 'p is not null and c is not null', extra='with synchronous_updates = true') as mv2, \
+             new_materialized_view(cql, table, '*', '(p,c),v', 'p is not null and c is not null and v is not null', extra='with synchronous_updates = true') as mv3:
+            N = 10
+            for i in range(N):
+                cql.execute(insert, [1, i])
+                cql.execute(insert, [2, i])
+
+            # Before the deletion, all N rows should exist in the base and the view
+            allN = list(range(N))
+            assert allN == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=1")]
+            assert allN == sorted([x.c for x in cql.execute(f"SELECT c FROM {mv1} WHERE p=1")])
+            assert allN == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=2")]
+            assert allN == sorted([x.c for x in cql.execute(f"SELECT c FROM {mv1} WHERE p=2")])
+            assert [1,2] == sorted([x.p for x in cql.execute(f"SELECT p FROM {mv2} WHERE c=0")])
+            assert [1,2] == sorted([x.p for x in cql.execute(f"SELECT p FROM {mv2} WHERE c=1")])
+            assert [1,2] == sorted([x.p for x in cql.execute(f"SELECT p FROM {mv2} WHERE c=2")])
+
+            cql.execute(f"DELETE FROM {table} WHERE p=1")
+
+            # After the deletion, all data should be gone from both base and view
+            assert [] == list(cql.execute(f"SELECT c FROM {table} WHERE p=1"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv1} WHERE p=1"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv2} WHERE p=1 ALLOW FILTERING"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv3} WHERE p=1 ALLOW FILTERING"))
+
+            assert allN == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=2")]
+            assert allN == sorted([x.c for x in cql.execute(f"SELECT c FROM {mv1} WHERE p=2")])
+
+            for i in range(N):
+                assert [2] == sorted([x.p for x in cql.execute(f"SELECT p FROM {mv2} WHERE c={i}")])
+
+            for i in range(N):
+                assert [v] == sorted([x.v for x in cql.execute(f"SELECT v FROM {mv3} WHERE p=2 AND c={i}")])
+
+# Test deleting a large partition when there is a view with the same partition
+# key, and verify that view updates metrics is increased by exactly 1. Deleting
+# a partition in this case is expected to generate one view update for deleting
+# the corresponding view partition by a partition tombstone.
+def test_base_partition_deletion_with_same_view_partition_key_metrics(cql, test_keyspace):
```

Comment:
Please add the "scylla_only" fixture to skip this test on Cassandra because it doesn't have the metrics. Sadly the ScyllaMetrics code you started to use doesn't take care of this.

Reply:
done